### DISCUSSION TITLE
How to mark shipping not required?

### DISCUSSION THREAD

**Discussion by DustinBracy** [saleor/saleor#11205] [Q&A] [answered]
I don't see a way to update the 'isShippingRequired' flag in checkout.  I'm using my warehouse id in the checkoutDeliveryMethodUpdate mutation instead of a shipping method id, but 'isShippingRequired' always returns true.  Do I need to mark products not shippable to make this change to false?  

I'm getting an error in checkout: Message: null value in column "base_shipping_price_amount" violates not-null constraint, but I've selected a warehouse as a pickup point rather than shipping on this checkout session so a shipping rate shouldn't be required.  

Any help would be much appreciated!

**Comment by korycins** [comment, marked as answer]
Hey @DustinBracy , 
This flag is controlled by productType. If you CheckoutLine has variant that comes from productType where shipping is required then, the shipping will be required for the line.
If Checkout has any line that requires the shipping, the query field `checkout.isShippinRequired` will return True.

You should use warehouseID in `checkoutDeliveryMethodUpdate` only when you want to use warehouse as a click & collect point. 

Some links that should be helpfull:
https://docs.saleor.io/docs/3.x/developer/stock-allocation#click-and-collect-warehouses
https://docs.saleor.io/docs/3.x/dashboard/configuration/warehouses#click--collect-related-settings
https://docs.saleor.io/docs/3.x/developer/checkout#shipping

Is shipping required for product type can be set from Saloer dashboard: dashboard->configuration->product types -> <find your product type> -> right corner:
https://demo.saleor.io/dashboard/product-types/UHJvZHVjdFR5cGU6MTg%3D


If you are getting not handled errors (like `Message: null value in column "base_shipping_price_amount" violates not-null constraint`), please create gh issue from bug template with steps to reproduce

**Reply by DustinBracy** [reply]
Thank you @korycins, I have submitted a github issue as I am still receiving the error after marking all products not shippable and re-trying with a new checkout.  I thought about making a new shipping method 'store pickup' with free shipping as a workaround, but another bug in the dashboard [(2546)](https://github.com/saleor/saleor-dashboard/issues/2546) prevents me from using a $0 shipping method.  I think I am stuck until I can find a resolution.
